### ISSUE TITLE
SAGE doesn't handle `chr` prefixes correctly

### ISSUE THREAD

**Issue by ckrushton** [LCR-BCCRC/lcr-modules#193]
By default, the "get_chromosomes" function in SAGE returns "chr" prefixed contig names if the reference genome has "38" in the name, or non-chr prefixed if it doesn't. This causes problems if we have data aligned chr-prefixed versions of GRCh37/hg19. Also, the associated SAGE reference files (hotspot VCF, high confidence regions BED) are hard-coded to be chr-prefixed for GRCh38, and non-chr-prefixed otherwise.

I have a modified (albeit overengineered) get_chromosomes function which handles "chr" prefixes, and includes all contigs for capture samples, but I'm not sure if we want to do something different

**Comment by Kdreval** [comment]
good thing you caught the get_chromosomes! Do you know if there is a way to just make a list from the content of the associated  main_chromosomes.txt file created by reference_files? I couldn't get it to work because functions cannot refer to sub workflows on a clean repository where no reference files were generated before, so came up with that implementation - clearly not really working. 
For the reference files - this is more complicated since they are provided by SAGE directly. I think they will either need to be lifted, or the version of the file created alongside the original when that genome build is used.

**Comment by ckrushton** [comment]
Ah that makes sense. I have a workaround where it just checks the .fai file to determine if the contigs are `chr` prefixed or not, since the index is specified as a rule input previously. Could we just modify `_input_references` to also include the `main_chromosomes.txt` file, to ensure that file exists?

**Comment by rdmorin** [comment]
I agree a configuration file should specify this mapping and all snakefiles should have a way to use that.

**Comment by Kdreval** [comment]
It was handled like this before:
```
def get_chromosomes(wildcards):
    path = reference_files("genomes/"+str(wildcards.genome_build)+"/genome_fasta/main_chromosomes.txt")
    input = str(path)
    with open(input, 'r') as chromosome_file:
        lines = chromosome_file.readlines()
        chromosomes=[]
        for x in lines:
               chromosomes= ",".join(chromosomes)
   return chromosomes
```
But this was not working regardless of the presence of main_chromosomes.txt file in the `_input_references` rule, because the function cannot see this file is generated by snakemake if you are starting on a fresh copy of the repo where no reference files are present. Maybe this can be adopted? How are you referring to fai file currently?

**Comment by ckrushton** [comment]
I'm referring to it in the same manner, but since the `fai` file is specified in `_input_references` it should be ok 99% of the time. Can we specify the `main_chromosomes.txt` file as an input to `run_sage`, which will ensure it exists
 even if its not directly called by `shell`?

**Comment by Kdreval** [comment]
Right, 99% of time because we already have reference files pre-generated. That error was picked up because someone else on GSC outside GAMBL tried to run it: https://morinlabsfu.slack.com/archives/C01FU4P6GLT/p1610819101002000 
It does not matter whether main_chromosomes or fai file is used, as long as you are restricting to autosomes+X. They will both give same output and both will error out if no reference files are available.

**Comment by ckrushton** [comment]
So what I ended up doing is modifying `run_sage` with
```
input:
    genome_chroms = reference_files("genomes/{genome_build}/genome_fasta/main_chromosomes.txt"),
    ...
```
Which should allow snakemake to use the reference workflow if necessary to create it

**Comment by Kdreval** [comment]
This is now addressed in the PR #215 and the buggy function is dropped in favour of using shell `paste` directly.

**Comment by Kdreval** [comment]
This is now addressed in the PR #215 and the buggy function is dropped in favour of using shell `paste` directly.
